### PR TITLE
Implement the Error Handling Split Process for "View Dependents"

### PR DESCRIPTION
**Pull request by cohnjesse**
## Description
Knowing what errors are possible and how they work can let us better inform the end user about what is happening with their data.

I only included a screenshot of the 400 series error response because the 500 series is harder to fake on localhost but I tested it and it works.

## Screenshots
[screencapture-localhost-3001-disability-view-dependents-2020-02-18-16_11_48.pdf](https://github.com/department-of-veterans-affairs/vets-website/files/4222440/screencapture-localhost-3001-disability-view-dependents-2020-02-18-16_11_48.pdf)

## Testing done
unit tests will be included in a separate branch

## Definition of done
- [x] Errors are shown when appropriate

### CODE REVIEW

File: src/applications/personalization/view-dependents/layouts/helpers.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      Monday-Friday, 8:00a.m.-8:00p.m. ET.
+    </p>
+  </>
+);
+
+export const infoFragment = (
+  <>
+    <p>
+      We can't find any dependents added to your disability award. If you are
+      eligable for VA disability compensation and you have a VA combined rating
```
Comment (micahchiang) [inline src/applications/personalization/view-dependents/layouts/helpers.jsx]: This should be spelled eligible.

File: src/applications/personalization/view-dependents/layouts/helpers.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      of 30%, you may be eligable for additional disability compensation for a
```
Comment (micahchiang) [inline src/applications/personalization/view-dependents/layouts/helpers.jsx]: This should be spelled eligible.

File: src/applications/personalization/view-dependents/layouts/helpers.jsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      spouse, child, and/or parent.
+    </p>
+    <a href="/disability/add-remove-dependent/">
+      Find out how to add a dependent to your disability claim
+    </a>
+  </>
+);
+
+export const breadcrumbLinks = [
+  <a href="/" aria-label="back to VA Home page" key="1">
```
Comment (micahchiang) [inline src/applications/personalization/view-dependents/layouts/helpers.jsx]: 'back` should be capitalized to remain consistent with the aria-labels below.

**Review by micahchiang — CHANGES_REQUESTED** [review]
Left a couple suggestions on accessibility and correct messaging.

File: src/applications/personalization/view-dependents/layouts/helpers.jsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+export const errorFragment = (
+  <>
+    <p>
+      Please refresh this page or check back later. You can also sign out of
+      VA.gov and try signing back into this page.
+    </p>
+    <p>
+      If you get this error again, please call the VA.gov help desk at
+      <a href="tel:8446982311">844-698-2311</a> (TTY:711). We're here
```
[the remark this answers is not in the document]
Comment (cohnjesse) [reply src/applications/personalization/view-dependents/layouts/helpers.jsx]: This should be addressed now. Question though, won't this make the screen reader read out the "dots"?

File: src/applications/personalization/view-dependents/layouts/ViewDependentsLayout.jsx [hunk 5]
```diff
@@ -10,42 +9,46 @@ import {
...
-      <a href="/disability/view-dependents/" key="4">
-        Your VA Dependents
-      </a>,
-    ];
+    let mainContent;
+
+    if (isServerError(this.props.error.code)) {
+      mainContent = (
+        <AlertBox
+          headline="We're sorry. Something went wrong on our end"
```
[the remark this answers is not in the document]
Comment (cohnjesse) [reply src/applications/personalization/view-dependents/layouts/ViewDependentsLayout.jsx]: This is addressed using the solution proposed

File: src/applications/personalization/view-dependents/layouts/ViewDependentsLayout.jsx [hunk 6]
```diff
@@ -10,42 +9,46 @@ import {
...
+          loading={this.props.loading}
+          onAwardDependents={this.props.onAwardDependents}
+          notOnAwardDependents={this.props.notOnAwardDependents}
+        />
+      );
+    }
 
-    const mainContent = (
+    const layout = (
       <div className="vads-l-grid-container vads-u-padding--0">
```
[the remark this answers is not in the document]
Comment (cohnjesse) [reply src/applications/personalization/view-dependents/layouts/ViewDependentsLayout.jsx]: Good catch, changed this to have the needed padding

File: src/applications/personalization/view-dependents/layouts/helpers.jsx (hunk 3 above)
Comment (cohnjesse) [reply src/applications/personalization/view-dependents/layouts/helpers.jsx]: This is fixed

File: src/applications/personalization/view-dependents/layouts/helpers.jsx (hunk 2 above)
Comment (cohnjesse) [reply src/applications/personalization/view-dependents/layouts/helpers.jsx]: This is fixed

File: src/applications/personalization/view-dependents/layouts/helpers.jsx (hunk 1 above)
Comment (cohnjesse) [reply src/applications/personalization/view-dependents/layouts/helpers.jsx]: This is fixed

File: src/applications/personalization/view-dependents/layouts/helpers.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (micahchiang) [reply src/applications/personalization/view-dependents/layouts/helpers.jsx]: I don't think so. Screen readers will typically detect a period and just pause. Similar to how someone reading out loud would.

**Review by micahchiang — APPROVED** [review]
LGTM. Thanks, @cohnjesse!

File: src/applications/personalization/view-dependents/components/ViewDependentsList/ViewDependentsList.jsx [hunk 7]
```diff
@@ -11,7 +11,7 @@ class ViewDependentsList extends Component {
       mainContent = (
         <LoadingIndicator message="Loading your dependents" setFocus />
       );
-    } else if (this.props.dependents) {
+    } else if (this.props.dependents && this.props.dependents.length > 0) {
```
Comment (bkjohnson) [inline src/applications/personalization/view-dependents/components/ViewDependentsList/ViewDependentsList.jsx]: I think optional chaining could be used here like so:

```js
     } else if (this?.props?.dependents.length > 0) { 
```

Here is a [link to some more info](https://dsva.slack.com/archives/CBU0KDSB1/p1567182149038900)

**Review by bkjohnson — APPROVED** [review]
Nothing to complain about, just found a possible opportunity to use optional chaining.

**Pull request merged**